composer/trainer/dist_strategy.py

```diff
@@ -369,78 +369,141 @@ def sync_hook(*args):
             if hasattr(obj, '_fsdp_wrap') and not bool(obj._fsdp_wrap):
                 continue
 
-            def _param_init_fn(module: torch.nn.Module) -> None:
-                # A dictionary of all tied parameter pointers to module names
+            # Rather than verifying these changes with older PyTorch versions, we are fixing forward here
+            if version.parse(torch.__version__) > version.parse('2.1.0'):
+                # A dictionary of all tied parameter pointers to (module, attr) tuples
                 tied_pointers = {}
 
                 # Goes through all modules finding which weights have the same pointers
```

Comment:
In what cases would different weights have the same pointers?

Reply:
It is common in transformer lms to have the input embedding and the output unembedding layers use the same weights. that is the main use i'm aware of, although likely weight tying may be used in other contexts.